brian-ndungu/ongeza_website
# fix(general): fix request lag on step where quote amount gets typed in and fetches quote amount

**Issue: {{OL-3}}**
Closes #3 

## Type
* Quick patch.

## Completed
* Added a debounce to the Premium and Cover amount inputs

## Reason
* To fix issue where page would show an outdated calculation from an amount that was calculated when the user was still typing.

## Benefits
*  A debouncer makes sure previous requests would get cancelled before if the user is still typing.

## Screenshots
* **Currently on master, not having the debounce logic causes the page to miss the correct amount to calculate.**
![no-debounce-on-master](https://user-images.githubusercontent.com/30522249/101245000-84b00800-3712-11eb-870f-40d75ddcc951.gif)

* **Proposed fix uses a debounce function to delay the calculation during keyup events, by cancelling previous function calls.**
![proposed-fix-with-debounce](https://user-images.githubusercontent.com/30522249/101245034-c476ef80-3712-11eb-8916-882a5196e89b.gif)

* **Proposed fix for Cover input**
![debounced-cover-input](https://user-images.githubusercontent.com/30522249/101245455-b9718e80-3715-11eb-9e2a-563a586bac04.gif)


## Verification Process
* Pull the branch.
* Open Chrome and open index.html
* Go to the Get a Quote section and verify on step two that when you type in the Premium or Cover inputs that the correct amounts show on the left.
